### PR TITLE
EPMDJ-4411: Correct redirects at finish scope

### PR DESCRIPTION
**Pull request by SMPlay**
Issue: EPMDJ-4411

### CODE REVIEW

**Review by RomanAchilov — APPROVED** [review]

File: src/pages/agent-full-page/coverage-plugin/scope/finish-scope-modal/finish-scope-modal.tsx [hunk 1]
```diff
@@ -31,12 +31,13 @@ export const FinishScopeModal = finishScopeModal(
...
     const [errorMessage, setErrorMessage] = React.useState('');
     const [ignoreScope, setIgnoreScope] = React.useState(false);
-
     const testsCount = scope
       ? (scope.coverage.byTestType || []).reduce((acc, { summary: { testCount = 0 } }) => acc + testCount, 0)
       : 0;
     const { pluginId = '' } = useParams<{ pluginId: string }>();
     const { push, location: { pathname = '' } } = useHistory();
+    const isActiveScopePage = scope?.id && pathname.includes(scope.id);
+    const isAllScopesPage = pathname.includes('scopes');
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/scope/finish-scope-modal/finish-scope-modal.tsx:40]: this is not true. scope info page includes `/scopes/`

File: src/pages/agent-full-page/coverage-plugin/scope/finish-scope-modal/finish-scope-modal.tsx [hunk 2]
```diff
@@ -31,12 +31,13 @@ export const FinishScopeModal = finishScopeModal(
     const { agentId, buildVersion } = usePluginState();
     const [errorMessage, setErrorMessage] = React.useState('');
     const [ignoreScope, setIgnoreScope] = React.useState(false);
-
     const testsCount = scope
       ? (scope.coverage.byTestType || []).reduce((acc, { summary: { testCount = 0 } }) => acc + testCount, 0)
       : 0;
     const { pluginId = '' } = useParams<{ pluginId: string }>();
     const { push, location: { pathname = '' } } = useHistory();
+    const isActiveScopePage = scope?.id && pathname.includes(scope.id);
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/scope/finish-scope-modal/finish-scope-modal.tsx:39]:
```suggestion
    const isScopeInfoPage = scope?.id && pathname.includes(scope.id);

```

**Review by pavyarov — CHANGES_REQUESTED** [review]

File: src/pages/agent-full-page/coverage-plugin/scope/finish-scope-modal/finish-scope-modal.tsx [hunk 3]
```diff
@@ -80,7 +81,7 @@ export const FinishScopeModal = finishScopeModal(
                     },
                     onError: setErrorMessage,
                   })({ prevScopeEnabled: !ignoreScope, savePrevScope: true });
-                  scope?.id && pathname.includes(scope.id)
+                  ((!testsCount && isScopeInfoPage) || (testsCount === 1 && recordTests && !isAllScopesPage))
```
Comment (RomanAchilov) [inline src/pages/agent-full-page/coverage-plugin/scope/finish-scope-modal/finish-scope-modal.tsx:84]: the condition is according to specification? It looks strange

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/pages/agent-full-page/coverage-plugin/scope/finish-scope-modal/finish-scope-modal.tsx (hunk 3 above)
Comment (SMPlay) [reply src/pages/agent-full-page/coverage-plugin/scope/finish-scope-modal/finish-scope-modal.tsx:84]: if the coverage is not fixed in the scope and tests are written, then the scope will be deleted and a redirect will occur. If I can test it differently then the condition looks better

Comment (RomanAchilov) [reply src/pages/agent-full-page/coverage-plugin/scope/finish-scope-modal/finish-scope-modal.tsx:84]: specification

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

**Review by pavyarov — APPROVED** [review]

**Pull request merged**